src/System.Text.Json/tests/JsonObjectTests.cs

```diff
@@ -0,0 +1,621 @@
+            {
+                { "name", "value" }
+            };
+
+            Assert.True(jsonObject.TryGetPropertyValue("name", out JsonNode property));
+            Assert.IsType<JsonString>(property);
+            Assert.Equal("value", property as JsonString);
+            Assert.False(jsonObject.TryGetPropertyValue("other", out property));
+            Assert.Null(property);
+        }
+
+        [Fact]
+        public static void TestGetJsonObjectPropertyThrows()
+        {
+            Assert.Throws<InvalidCastException>(() =>
+            {
+                var jsonObject = new JsonObject()
+                {
+                    { "name", "value" }
+                };
+
+                jsonObject.GetJsonObjectPropertyValue("name");
+            });
+        }
+
+        [Fact]
+        public static void TestTryGetObjectPropertyFails()
+        {
+            var jsonObject = new JsonObject()
+            {
+                { "name", "value" }
+            };
+
+            Assert.False(jsonObject.TryGetJsonObjectPropertyValue("name", out JsonObject property));
+            Assert.Null(property);
+
+            Assert.False(jsonObject.TryGetJsonObjectPropertyValue("other", out property));
+            Assert.Null(property);
+        }
+
+        [Fact]
+        public static void TestArgumentNullValidation()
+        {
+            Assert.Throws<ArgumentNullException>(() => new JsonObject().Add(null, ""));
+            Assert.Throws<ArgumentNullException>(() => new JsonObject().Add(null, new JsonObject()));
+            Assert.Throws<ArgumentNullException>(() => new JsonObject().Add(null, (byte)17));
+            Assert.Throws<ArgumentNullException>(() => new JsonObject().Add(null, (short)17));
+            Assert.Throws<ArgumentNullException>(() => new JsonObject().Add(null, 17));
+            Assert.Throws<ArgumentNullException>(() => new JsonObject().Add(null, (long)17));
+            Assert.Throws<ArgumentNullException>(() => new JsonObject().Add(null, 3.14f));
+            Assert.Throws<ArgumentNullException>(() => new JsonObject().Add(null, 3.14));
+            Assert.Throws<ArgumentNullException>(() => new JsonObject().Add(null, decimal.One));
+            Assert.Throws<ArgumentNullException>(() => new JsonObject().Add(null, (sbyte)17));
+            Assert.Throws<ArgumentNullException>(() => new JsonObject().Add(null, (ushort)17));
+            Assert.Throws<ArgumentNullException>(() => new JsonObject().Add(null, (uint)17));
+            Assert.Throws<ArgumentNullException>(() => new JsonObject().Add(null, (ulong)17));
+            Assert.Throws<ArgumentNullException>(() => new JsonObject().Add(new KeyValuePair<string, JsonNode>(null, new JsonObject())));
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                var property = new KeyValuePair<string, JsonNode>(null, new JsonObject());
+                new JsonObject().AddRange(new List<KeyValuePair<string, JsonNode>>() { property });
```

Comment:
How about one where we have non-null entries within the List of KVP, with one of the entries being null.